File: my-folder/0041-first-missing-positive/solution.java
class Solution {
    public int firstMissingPositive(int[] nums) {
        boolean one=false;
        int n=nums.length;
        for(int i=0;i<n;i++)
        {
            if(nums[i]==1)
                one=true;
             if(nums[i]<1 || nums[i]>n)
                nums[i]=1;
            
        }
        if(one==false)
            return 1;
        for(int i=0;i<n;i++)
        {
          int element=Math.abs(nums[i]);
          nums[element-1]=-Math.abs(nums[element-1]);
        }
        
        for(int i=0;i<n;i++)
        {
          if(nums[i]>0)
              return i+1;
        }
        
        return n+1;
        
        
    }
}

File: my-folder/0041-first-missing-positive/solution.kt
class Solution {
    fun firstMissingPositive(nums: IntArray): Int {
        for(i in nums.indices)
        {
            while(nums[i] in 1..nums.size && nums[nums[i]-1]!=nums[i]){
                val correctIndex=nums[i]-1
                val temp=nums[correctIndex]
                nums[correctIndex]=nums[i]
                nums[i]=temp
            }   
        }
         for(i in nums.indices)
         {
            if(nums[i]!=i+1)
               return i+1
         }
         return nums.size+1

    }
}
